components/clarinet-cli/src/frontend/cli.rs

```diff
@@ -579,7 +584,26 @@ pub fn main() {
         }
     };
 
+    let mut global_settings = String::new();
+    let global_settings_default = GlobalSettings {
+        disable_hints: false,
+        enable_telemetry: None,
+    };
+    // This is backwards compatible with ENV var setting as well as the new ~/.clarinet/Settings.toml
     let hints_enabled = env::var("CLARINET_DISABLE_HINTS") != Ok("1".into());
+    let home_dir = dirs::home_dir().expect("Unable to get home directory");
+    let path = home_dir.join(".clarinet/Settings.toml");
+
+    let global_settings: GlobalSettings = if path.exists() {
+        let mut file = File::open(&path).expect("Unable to open the file");
+        file.read_to_string(&mut global_settings)
+            .expect("Unable to read the file");
```

Comment:
I think it would be better to log an warning instead of panicking whenever something unexpected happens.
Also, the error message could be more explicit: `"the file"` -> `"the ~/.clarinet/Settings.toml file"`

Reply:
Used format_warn for this
![Screenshot 2023-10-24 at 6 05 15 PM](https://github.com/hirosystems/clarinet/assets/1235263/7bc1a4e4-cede-4b91-813e-af0d9ba4aa06)